### PR TITLE
fix(client, cli): patch for data proxy fixes

### PR DESCRIPTION
**Pull request by millsp**
Related https://github.com/prisma/docs/pull/3274

### CODE REVIEW

File: packages/cli/src/Generate.ts [hunk 1]
```diff
@@ -138,7 +138,7 @@ ${chalk.bold('Examples')}
         printDownloadProgress: !watchMode,
         version: enginesVersion,
         cliVersion: pkg.version,
-        dataProxy: !!args['--data-proxy'],
+        dataProxy: !!args['--data-proxy'] || !!process.env.PRISMA_CLI_GENERATE_DATAPROXY,
```
Comment (janpio) [inline packages/cli/src/Generate.ts:141]: Are we testing `--data-proxy` here in any way? Can we add one for `PRISMA_CLI_GENERATE_DATAPROXY` then as well? (Fine if not, then we need to via ecosystem-tests)

**Review by janpio — APPROVED** [review]

File: packages/cli/src/Generate.ts (hunk 1 above)
Comment (millsp) [reply packages/cli/src/Generate.ts:141]: Outstanding task for post GA, only tested via ecotests right now.

Comment (millsp) [reply packages/cli/src/Generate.ts:141]: (...but `--data-proxy` is tested via `getGenerators`)

Comment (janpio) [reply packages/cli/src/Generate.ts:141]: So in theory we could replicate that test with `--data-proxy` but with an env var?

**Comment by janpio** [comment]
Note: Needs to be merged back to `main` later.

**Pull request merged**